components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,352 @@
+		return error('Not a valid type!')
+	end
+
+	self.opponentType = self.args.type.type
+
+	return self
+end
+
+function PrizePool:readInput(args)
+	self.options = self:_readStandardOptions(args)
+	self.prizes = self:_readPrizes(args)
+	self.placements = self:_readPlacements(args)
+
+	if self:_hasUsdPrizePool() then
+		self:setOption('showUSD', true)
+		self:addPrize('USD', 1)
+	end
+
+	return self
+end
+
+function PrizePool:create()
+	-- TODO
+	mw.logObject(self)
+	return ''
+end
+
+function PrizePool:_readStandardOptions(args)
+	local options = {}
+
+	self:setOption('autoUSD', Logic.nilOr(Logic.readBoolOrNil(args.autousd), true), options)
+	self:setOption('showUSD', false, options)
+	self:setOption('prizeSummary', Logic.nilOr(Logic.readBoolOrNil(args.prizesummary), true), options)
+	self:setOption('exchangeInfo', Logic.nilOr(Logic.readBoolOrNil(args.exchangeinfo), true), options)
+	self:setOption('storeSmw', Logic.nilOr(Logic.readBoolOrNil(args.storesmw), true), options)
+	self:setOption('storeLpdb', Logic.nilOr(Logic.readBoolOrNil(args.storelpdb), true), options)
+
+	return options
+end
+
+function PrizePool:_readPrizes(args)
+	local prizes = {}
+
+	for prizeEnum in pairs(prizeTypes) do
+		local prizeDatum = prizeData[prizeTypes[prizeEnum]]
+		local fieldName = prizeDatum.header
+		if fieldName then
+			args[fieldName .. '1'] = args[fieldName .. '1'] or args[fieldName]
+			for _, prizeValue, index in Table.iter.pairsByPrefix(args, fieldName) do
+				local data = prizeDatum.headerParse(self, prizeValue, args, index)
+				self:addPrize(prizeEnum, index, data, prizes)
+			end
+		end
+	end
+
+	return prizes
+end
+
+function PrizePool:_readPrizeRewards(args)
+	local rewards = {}
+
+	for prizeEnum in pairs(prizeTypes) do
+		local prizeDatum = prizeData[prizeTypes[prizeEnum]]
+		local fieldName = prizeDatum.row
+		if fieldName then
+			args[fieldName .. '1'] = args[fieldName .. '1'] or args[fieldName]
+			for _, rewardValue, index in Table.iter.pairsByPrefix(args, fieldName) do
+				rewards[prizeEnum .. index] = prizeDatum.rowParse(self, rewardValue, args, index)
+			end
+		end
+	end
+
+	return rewards
+end
+
+function PrizePool:_readSpecialData(args)
+	local data = {}
+
+	for enum in pairs(specialDataTypes) do
+		local type = specialData[specialDataTypes[enum]]
+		local fieldName = type.field
+		if args[fieldName] then
+			data[enum] = type.parse(self, args[fieldName], args)
+		end
+	end
+
+	return data
+end
+
+function PrizePool:_readPlacements(args)
+	local placements = {}
+
+	local currentPlace = 0
+	for placementIndex = 1, math.huge do
+		local placementInput = Json.parseIfString(args[placementIndex])
+		if not placementInput then
+			break
+		end
+
+		local placement = {}
+		-- Parse place, explicit
+		if placementInput.place then
+			local places = Table.mapValues(mw.text.split(placementInput.place, '-'), tonumber)
+			placement.placeStart = places[1]
+			placement.placeEnd = places[2] or places[1]
```

Comment:
Is there a world where `places` < 2?

Reply:
Yes, `|place=1`

Reply:
Oh, misread. Can even `places[1]` be null?

Reply:
yes with invalid input

Reply:
Move an assert that was elsewhere into here.